sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java

```diff
@@ -1302,7 +1384,68 @@ public Response<PathProperties> readToFileWithResponse(String filePath, FileRang
                     .setRequestConditions(Transforms.toBlobRequestConditions(requestConditions))
                     .setRetrieveContentRangeMd5(rangeGetContentMd5).setOpenOptions(openOptions), timeout,
                 context);
-            return new SimpleResponse<>(response, Transforms.toPathProperties(response.getValue()));
+            return new SimpleResponse<>(response, Transforms.toPathProperties(response.getValue(), response));
+        }, LOGGER);
+    }
+
+    /**
+     * Reads the entire file into a file specified by the path.
+     *
+     * <p>By default the file will be created and must not exist, if the file already exists a
+     * {@link FileAlreadyExistsException} will be thrown. To override this behavior, provide appropriate
+     * {@link OpenOption OpenOptions} </p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <!-- src_embed com.azure.storage.file.datalake.DataLakeFileClient.readToFileWithResponse#ReadToFileOptions-Duration-Context -->
+     * <pre>
+     * ReadToFileOptions options = new ReadToFileOptions&#40;&#41;;
+     * options.setRange&#40;new FileRange&#40;1024, 2048L&#41;&#41;;
+     * options.setDownloadRetryOptions&#40;new DownloadRetryOptions&#40;&#41;.setMaxRetryRequests&#40;5&#41;&#41;;
+     * options.setOpenOptions&#40;new HashSet&lt;&gt;&#40;Arrays.asList&#40;StandardOpenOption.CREATE_NEW,
+     *         StandardOpenOption.WRITE, StandardOpenOption.READ&#41;&#41;&#41;; &#47;&#47;Default options
+     * options.setParallelTransferOptions&#40;new ParallelTransferOptions&#40;&#41;.setBlockSizeLong&#40;4L * Constants.MB&#41;&#41;;
+     * options.setDataLakeRequestConditions&#40;null&#41;;
+     * options.setRangeGetContentMd5&#40;false&#41;;
+     *
+     * client.readToFileWithResponse&#40;options, timeout, new Context&#40;key2, value2&#41;&#41;;
+     * System.out.println&#40;&quot;Completed download to file&quot;&#41;;
+     * </pre>
+     * <!-- end com.azure.storage.file.datalake.DataLakeFileClient.readToFileWithResponse#ReadToFileOptions-Duration-Context -->
+     *
+     * @param options {@link ReadToFileOptions}
+     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @return A response containing the file properties and metadata.
+     * @throws UncheckedIOException If an I/O error occurs.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<PathProperties> readToFileWithResponse(ReadToFileOptions options, Duration timeout, Context context) {
+        Context newContext;
+        options = options == null ? new ReadToFileOptions() : options;
```

Comment:
I think the comments I left in the other location setting this could apply everywhere